include/NovelRT/Input/InteractionService.h

```diff
@@ -13,22 +13,23 @@ namespace NovelRT::Input {
     friend class Windowing::WindowingService; //I get this looks weird but its because GLFW treats the window as this system as well as the window.
 
   private:
-    NovelRunner* const _runner;
+    static inline const uint32_t INPUT_BUFFER_COUNT = 2;
 
+    uint32_t _previousBufferIndex;
```

Comment:
Why do you need to track previous and current, rather than just current and implicitly determining previous?

Reply:
because implicitly determining previous everywhere is annoying and I prefer to have an actual cached index for it.

Reply:
Why not just track a ref to the right index and update it when you need to calculate the current?
Or centralize the logic in a single place? It's really just `(current == 0) ? count - 1 : current - 1`

Reply:
I already do provide the previous index when I calculate the current, they get cached at the same time. I'm unsure what benefit this would have.

Reply:
It avoids bloating the struct by an additional 4 bytes for something that is trivially computable.

Reply:
But then I'd need to compute it multiple times in a single pass? Sorry, I'm still not understanding this. I would've thought just calculating and caching ones for the entire pass of the input system to use would have been better than calculating it once in the FSM, calculating it once in `processKeys` and calculating it wherever else its needed.

Reply:
Why would it need to be computed multiple times? Maybe this discussion would be better for discord?

Reply:
Sure, I'll move over there.